api/admin/controller.py

```diff
@@ -2813,6 +2817,22 @@ def sitewide_registration(self, integration, do_get=HTTP.debuggable_get,
         shared_secret = encryptor.decrypt(base64.b64decode(shared_secret))
         integration.password = unicode(shared_secret)
 
+    def sitewide_registration_document(self, private_key):
```

Comment:
This is called sitewide registration even though it's only used by the metadata wrangler. If we did add another component that circ managers needed to register with, I'm not sure we'd want to use this.

Reply:
Assuming we get to control the registration protocol, I would like to reuse this document. The trust boundaries would be the same as with the metadata wrangler. We might have to add extra things to the JWT or to the resulting document but that would be a pretty easy change. Either way, we can handle this when it comes up.